src/main/java/org/schemaspy/SchemaAnalyzer.java

```diff
@@ -328,7 +328,12 @@ private void generateHtmlDoc(Config config, boolean useVizJS, ProgressListener p
         progressListener.graphingSummaryProgressed();
 
         boolean showDetailedTables = tables.size() <= config.getMaxDetailedTables();
-        final boolean includeImpliedConstraints = commandLineArguments.withImpliedRelationships();
+
+        // getting implied constraints has a side-effect of associating the parent/child tables, so don't do it
+        // here unless they want that behavior
+        List<ImpliedForeignKeyConstraint> impliedConstraints = new ImpliedConstraintsFinder(tables, commandLineArguments.withImpliedRelationships()).find();
```

Comment:
`ImpliedConstraintsFinder` should not take a boolean. Instead, the command line argument should determine what your object graph looks like.